File: bogo_sort.py
import random

def is_sorted(values: list) -> bool:
    for i in range(1, len(values)):
        if values[i - 1] > values[i]:
            return False
    return True

# 1.7642974853515625e-05 for 1
# 3.886222839355469e-05 for 2
# 5.054473876953125e-05 for 4
# 0.0959320068359375 for 8
def bogo_sort(values: list):
    while not is_sorted(values):
        random.shuffle(values)

File: src/bogo_sort.rs
// 0.000005871 for 1
// 0.000034767 for 2
// 0.000016156 for 4
// 0.007449113 for 8
pub fn bogo_sort<T: PartialOrd>(values: &mut Vec<T>){
    while !values.is_sorted() {
        let mut new_values = Vec::with_capacity(values.len());
        for _ in 0..values.len(){
            let index = rand::random::<usize>() % values.len();
            new_values.push(values.swap_remove(index));
        }
        *values = new_values;
    }
}
